### PR TITLE
Bring us back up to date with Upstream

### PR DESCRIPTION
**Pull request by MaggieFero**
There are some nice security changes here, and a few other little bits as well.

### CODE REVIEW

**Comment by MaggieFero** [comment]
We're going to hold off on merging this until we've got the branch with our security upgrades running in prod, I think.

**Review by phildini — APPROVED** [review]

**Comment by MaggieFero** [comment]
Actually, let's not do this for now, because we're submitting a PR to upstream right now and we'll catch up once it's merged.

**Comment by MaggieFero** [comment]
Hey! It's time! I'll open a new PR for this, and I expect it to be #50

**Pull request closed without merging**
